String datatype not compiling with stm32f103 boards

It seems strange to me why didn't this code compile. 

```
#include <Arduino.h>
String test="";
void setup() {
    // put your setup code here, to run once:
}

void loop() {
    // put your main code here, to run repeatedly:
}
```

This is the error i get while i compile this.
```
> Executing task: platformio run <

Processing genericSTM32F103C8 (platform: ststm32; board: genericSTM32F103C8; framework: arduino)
------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
PLATFORM: ST STM32 > STM32F103C8 (20k RAM. 64k Flash)
SYSTEM: STM32F103C8T6 72MHz 20KB RAM (64KB Flash)
DEBUG: CURRENT(blackmagic) EXTERNAL(blackmagic, jlink, stlink)
Converting main.ino
Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF MODES: FINDER(chain) COMPATIBILITY(soft)
Collected 36 compatible libraries
Scanning dependencies...
No dependencies
Compiling .pioenvs/genericSTM32F103C8/src/main.ino.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduinoVariant/board.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduinoVariant/wirish/boards.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduinoVariant/wirish/boards_setup.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduinoVariant/wirish/start.S.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduinoVariant/wirish/start_c.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduinoVariant/wirish/syscalls.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/HardwareSerial.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/HardwareTimer.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/IPAddress.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/Print.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/Stream.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/WString.cpp.o
Archiving .pioenvs/genericSTM32F103C8/libFrameworkArduinoVariant.a
Indexing .pioenvs/genericSTM32F103C8/libFrameworkArduinoVariant.a
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/avr/dtostrf.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/cxxabi-compat.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/ext_interrupts.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/hooks.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/itoa.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/adc.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/adc_f1.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/bkp_f1.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/dac.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/dma.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/dma_f1.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/exc.S.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/exti.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/exti_f1.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/flash.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/fsmc_f1.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/gpio.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/gpio_f1.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/i2c.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/i2c_f1.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/iwdg.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/nvic.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/pwr.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/rcc.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/rcc_f1.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/spi.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/spi_f1.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/stm32f1/performance/isrs.S.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/stm32f1/performance/vector_table.S.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/systick.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/timer.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/timer_f1.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/usart.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/usart_f1.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/usart_private.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/usb/stm32f1/usb.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/usb/stm32f1/usb_cdcacm.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/usb/stm32f1/usb_reg_map.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/usb/usb_lib/usb_core.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/usb/usb_lib/usb_init.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/usb/usb_lib/usb_mem.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/usb/usb_lib/usb_regs.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/libmaple/util.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/main.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/new.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/pwm.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/sdio.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/stm32f1/util_hooks.c.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/stm32f1/wiring_pulse_f1.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/stm32f1/wirish_debug.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/stm32f1/wirish_digital_f1.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/tone.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/usb_serial.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/wirish_analog.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/wirish_digital.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/wirish_math.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/wirish_shift.cpp.o
Compiling .pioenvs/genericSTM32F103C8/FrameworkArduino/wirish_time.cpp.o
Archiving .pioenvs/genericSTM32F103C8/libFrameworkArduino.a
Indexing .pioenvs/genericSTM32F103C8/libFrameworkArduino.a
Linking .pioenvs/genericSTM32F103C8/firmware.elf
/home/gopi/.platformio/packages/toolchain-gccarmnoneeabi/bin/../lib/gcc/arm-none-eabi/7.2.1/../../../../arm-none-eabi/lib/thumb/v7-m/libnosys.a(sbrk.o): In function `_sbrk':
sbrk.c:(.text._sbrk+0x18): undefined reference to `end'
collect2: error: ld returned 1 exit status
*** [.pioenvs/genericSTM32F103C8/firmware.elf] Error 1
==================================================== [ERROR] Took 11.96 seconds ====================================================
The terminal process terminated with exit code: 1
```

If i comment out that string which is initialized, the program compiles without any error. Can someone help me solving this error.

--- a/builder/frameworks/arduino/maple/stm32f1.py
+++ b/builder/frameworks/arduino/maple/stm32f1.py
@@ -182,7 +182,7 @@ env.Append(
 
     LIBPATH=[join(FRAMEWORK_DIR, "variants", variant, "ld")],
 
-    LIBS=["c", "gcc", "m"]
+    LIBS=["c", "gcc", "m", "c"]
 )
 
 # copy CCFLAGS to ASFLAGS (-x assembler-with-cpp mode)

